setup.py

```diff
@@ -23,6 +24,8 @@ def get_version():
 def get_requirements():
     """Reads the requirements file."""
     requirements = read("requirements.txt")
+    if sys.version_info < (3, 5):
```

Comment:
If we want to pull in only for specific Python versions, we should indicate that using environment markers[1] in `requirements.txt`, not by programmatic modification in `setup.py`.

That said, thinking about this, I think the better answer might be for us to just drop the requirement altogether.

The only thing we use `typing` for is typehint annotations, and those are only evaluated when we run `mypy`. Because we support Python 2.7, those annotations have to be in comments, so they are never evaluated at runtime.
Finally, because of issues with CPython `3.5.0` and `3.5.1`, we have to wrap all of our imports from `typing` in try/catch blocks[2] to avoid breaking when used with those patch versions.

Because of all of this, I think that we can just drop the `typing` requirement entirely and simply make sure that the mypy tox environments are run with greater than `3.5.1`.

[1] https://www.python.org/dev/peps/pep-0508/#environment-markers
[2] https://github.com/aws/aws-encryption-sdk-cli/blob/master/src/aws_encryption_sdk_cli/key_providers.py#L22-L26